src/main/java/com/pinterest/secor/parser/ThriftMessageParser.java

```diff
@@ -52,9 +53,11 @@ public String getFieldName() {
         }
     }
 
-    public ThriftMessageParser(SecorConfig config) {
+    public ThriftMessageParser(SecorConfig config) throws InstantiationException, IllegalAccessException, ClassNotFoundException {
         super(config);
-        mDeserializer = new TDeserializer();
+        String factoryClassName = mConfig.getThriftProtocolClass().concat("$Factory");
+		TProtocolFactory protocolFactory = ((Class<? extends TProtocolFactory>) Class.forName(factoryClassName)).newInstance();
+        mDeserializer = new TDeserializer(protocolFactory);
```

Comment:
This is a new API with procotolFactory, is this backward compatible with the previous code?